File: Problems_and_Solutions/day-9/cpp/solution 3.cpp
class Solution {
  public:
    Node *insertAtEnd(Node *head, int x) {
        // Code here
        Node* prev=head;
        Node* cur=new Node(x);
        if(prev==NULL)
        {
            head=cur;
        }
        else{
            while(prev->next!=NULL)
                prev=prev->next;
            prev->next=cur;
        }
        
        return head;
    }
};

File: Problems_and_Solutions/day-9/Java/Solution3.java
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class Solution3 {
    public static Node insertAtEnd(Node head, int x) {
        Node newNode = new Node(x);
        if (head == null) {
            return newNode;
        }
        Node curr = head;
        while (curr.next != null) {
            curr = curr.next;
        }
        curr.next = newNode;
        return head;
    }

    public static void printList(Node head) {
        Node curr = head;
        while (curr != null) {
            System.out.print(curr.data + " ");
            curr = curr.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Node head = null;
        head = insertAtEnd(head, 10);
        head = insertAtEnd(head, 20);
        head = insertAtEnd(head, 30);
        head = insertAtEnd(head, 40);
        printList(head);
    }
}
